src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs

```diff
@@ -433,6 +433,18 @@ private void OnUpdateGraphCompleted(AsyncTask task)
             var updateTask = (UpdateGraphAsyncTask)task;
             var messages = new Dictionary<Guid, string>();
 
+            // Mark all modified nodes as being updated.
+            // 
+            // In addition to marking modified nodes as being updated, their 
+            // warning states are cleared (which include the tool-tip).
+            foreach (var modifiedNode in updateTask.ModifiedNodes)
```

Comment:
Sorry for the late respond, @ke-yu! Yes as @aparajit-pratap pointed out, the documentation for `OnUpdateGraphCompleted` says this (so it still runs on the scheduler thread):

```csharp
/// <summary>
/// This callback method is invoked in the context of ISchedulerThread 
/// when UpdateGraphAsyncTask is completed.
/// </summary>
/// <param name="task">The original UpdateGraphAsyncTask instance.</param>
```